src/clib/pioc_support.c

```diff
@@ -742,25 +901,14 @@ int PIOc_freedecomp(int iosysid, int ioid)
     /* If async is in use, and this is not an IO task, bcast the parameters. */
     if (ios->async)
     {
-        if (!ios->ioproc)
-        {
-            int msg = PIO_MSG_FREEDECOMP; /* Message for async notification. */
-
-            if (ios->compmaster == MPI_ROOT)
-                mpierr = MPI_Send(&msg, 1, MPI_INT, ios->ioroot, 1, ios->union_comm);
+        int msg = PIO_MSG_FREEDECOMP; /* Message for async notification. */
 
-            if (!mpierr)
-                mpierr = MPI_Bcast(&iosysid, 1, MPI_INT, ios->compmaster, ios->intercomm);
-            if (!mpierr)
-                mpierr = MPI_Bcast(&ioid, 1, MPI_INT, ios->compmaster, ios->intercomm);
-            LOG((2, "PIOc_freedecomp iosysid = %d ioid = %d", iosysid, ioid));
+        PIO_SEND_ASYNC_MSG(ios, msg, &ret, iosysid, ioid);
```

Comment:
C macros like this break debugging for very little benefit. Whatever can be expressed as a pre-processor macro can also be expressed very similarly as a C function.